README.md

```diff
@@ -41,6 +39,45 @@ asdf uses a `.tool-versions` file for auto-switching between software versions.
 legacy_version_file = yes
 ```
 
+## Partial and codename versions
+
+Many version managers allow you to use partial versions (e.g. `v10`) or NodeJS codenames (e.g. `lts/hydrogen`) in version files, which are resolved at runtime. However, this can be risky as it is not guaranteed that all developers will use the same version, leading to non-reproducibility. In `asdf`, we prioritize reproducibility, so you cannot use partial versions or codenames in a `.tool-versions` file.
+
+To address this, we offer an escape hatch for legacy version files. If you are comfortable with non-reproducibility issues, you can provide your own resolution script as an environment variable `ASDF_NODEJS_RESOLVE_VERSION_CMD`. You can export this variable from your shell rc file and it will become the default behavior.
```

Comment:
This only works if asdf's `legacy_version_file` setting is set to `yes`? It might be good to state that explicitly here.